shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -892,14 +1048,26 @@ public boolean shouldDestroyEngineWithHost() {
   /**
    * The Dart entrypoint that will be executed as soon as the Dart snapshot is loaded.
    *
-   * <p>This preference can be controlled by setting a {@code <meta-data>} called {@link
-   * FlutterActivityLaunchConfigs#DART_ENTRYPOINT_META_DATA_KEY} within the Android manifest
-   * definition for this {@code FlutterActivity}.
+   * <p>This preference can be controlled with 2 methods:
+   *
+   * <ol>
+   *   <li>Pass a boolean as {@link FlutterActivityLaunchConfigs#EXTRA_DART_ENTRYPOINT} with the
+   *       launching {@code Intent}, or
+   *   <li>Set a {@code <meta-data>} called {@link
+   *       FlutterActivityLaunchConfigs#DART_ENTRYPOINT_META_DATA_KEY} within the Android manifest
+   *       definition for this {@code FlutterActivity}
+   * </ol>
+   *
+   * If both preferences are set, the {@code Intent} preference takes priority.
    *
    * <p>Subclasses may override this method to directly control the Dart entrypoint.
    */
   @NonNull
   public String getDartEntrypointFunctionName() {
+    if (getIntent().hasExtra(EXTRA_DART_ENTRYPOINT)) {
+      return getIntent().getStringExtra(EXTRA_DART_ENTRYPOINT);
+    }
```

Comment:
Why did we have to add a second way?

Reply:
EngineGroup can specify an entrypoint, so we can pass it in the intent. Previously the entrypoint was set in MeteData, which was 'main 'by default

Reply:
So was this a convenience of not having to use MetaData or is it allowing us to do something we couldn't before?

Reply:
Yes, previously it was only possible with MeteData (I think because previously we would only create one Engine),  In EngineGroup it's possible to use Intent directly.